algorithms/kernel/dbscan/oneapi/dbscan_dense_default_batch_ucapi_impl.i

```diff
@@ -0,0 +1,511 @@
+            DAAL_CHECK_STATUS_VAR(st);
+            pos++;
+            DAAL_CHECK_STATUS_VAR(ntData->releaseBlockOfRows(dataRows));
+        }
+        if (result)
+        {
+            return Status(services::ErrorMemoryCopyFailedInternal);
+        }
+    }
+
+    return Status();
+}
+
+template <typename algorithmFPType>
+Status DBSCANBatchKernelUCAPI<algorithmFPType>::compute(const NumericTable * x, const NumericTable * ntWeights, NumericTable * ntAssignments,
+                                                        NumericTable * ntNClusters, NumericTable * ntCoreIndices, NumericTable * ntCoreObservations,
+                                                        const Parameter * par)
+{
+    Status s;
+    auto & context        = Environment::getInstance()->getDefaultExecutionContext();
+    auto & kernel_factory = context.getClKernelFactory();
+
+    const algorithmFPType epsilon         = par->epsilon;
+    const algorithmFPType minObservations = par->minObservations;
+    const uint32_t minkowskiPower         = (uint32_t)2.0;
+    algorithmFPType epsP                  = 1.0;
+    for (uint32_t i = 0; i < minkowskiPower; i++) epsP *= epsilon;
+
+    NumericTable * ntData = const_cast<NumericTable *>(x);
+    if (ntData->getNumberOfRows() > static_cast<size_t>(UINT_MAX) || ntData->getNumberOfColumns() > static_cast<size_t>(UINT_MAX))
+    {
+    }
+    const uint32_t nRows = ntData->getNumberOfRows();
+    const uint32_t dim   = ntData->getNumberOfColumns();
+
+    BlockDescriptor<algorithmFPType> dataRows;
+    ntData->getBlockOfRows(0, nRows, readOnly, dataRows);
+    auto data = dataRows.getBuffer();
+
+    BlockDescriptor<int> assignRows;
+    DAAL_CHECK_STATUS_VAR(ntAssignments->getBlockOfRows(0, nRows, writeOnly, assignRows));
+    auto assignments = daal::oneapi::internal::UniversalBuffer(assignRows.getBuffer());
+    context.fill(assignments, undefined, &s);
+    DAAL_CHECK_STATUS_VAR(s);
+
+    DAAL_OVERFLOW_CHECK_BY_MULTIPLICATION(uint32_t, _queueBlockSize, nRows);
+
+    auto queueBlockDistances = context.allocate(TypeIds::id<algorithmFPType>(), _queueBlockSize * nRows, &s);
+    DAAL_CHECK_STATUS_VAR(s);
+    auto singlePointDistances = context.allocate(TypeIds::id<algorithmFPType>(), nRows, &s);
+    DAAL_CHECK_STATUS_VAR(s);
+    auto queue = context.allocate(TypeIds::id<int>(), nRows, &s);
+    DAAL_CHECK_STATUS_VAR(s);
+    auto isCore = context.allocate(TypeIds::id<int>(), nRows, &s);
+    DAAL_CHECK_STATUS_VAR(s);
+    auto countersTotal = context.allocate(TypeIds::id<int>(), _chunkNumber, &s);
+    DAAL_CHECK_STATUS_VAR(s);
+    auto countersNewNeighbors = context.allocate(TypeIds::id<int>(), _chunkNumber, &s);
+    DAAL_CHECK_STATUS_VAR(s);
+    auto chunkOffests = context.allocate(TypeIds::id<int>(), _chunkNumber, &s);
+    DAAL_CHECK_STATUS_VAR(s);
+
+    context.fill(isCore, 0, &s);
+    DAAL_CHECK_STATUS_VAR(s);
```

Comment:
You can make all of these arrays as fields of the class and initialize them in different method